### ISSUE TITLE
Display of data flow

### ISSUE THREAD

**Issue by marknfischer** [inspire/woocommerce-gateway-value.io#4]
The more I use this plugin, the more strange it feels to show the receipt page, and then the modal.  Is there a .js way to stop the form BEFORE it is submitted, and build the modal over the checkout page rather than the receipt page?  I know it might be a major refactor, but the way we have it now doesn't fit in the norms of ecommerce.  Needs a refactor.

**Comment by shulmang** [comment]
Yeah, from a user experience perspective it is definitely awkward. I'm 
sure we've got some options.
